yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -585,23 +642,50 @@ func (yb *TargetYugabyteDB) IsNonRetryableCopyError(err error) bool {
 	return err != nil && utils.InsensitiveSliceContains(NonRetryCopyErrors, err.Error())
 }
 
-func (yb *TargetYugabyteDB) ExecuteBatch(batch []*Event) error {
-	var err error
-	for i := 0; i < len(batch); i++ {
-		event := batch[i]
-		stmt := event.GetSQLStmt(yb.tconf.Schema)
-		log.Debug(stmt)
-		err = yb.connPool.WithConn(func(conn *pgx.Conn) (bool, error) {
-			_, err := conn.Exec(context.Background(), stmt)
+func (yb *TargetYugabyteDB) ExecuteBatch(migrationUUID uuid.UUID, batch EventBatch) error {
+	err := yb.connPool.WithConn(func(conn *pgx.Conn) (retry bool, err error) {
+		ctx := context.Background()
+		tx, err := conn.BeginTx(ctx, pgx.TxOptions{})
+		if err != nil {
+			return false, fmt.Errorf("error creating tx: %w", err)
+		}
+		defer func() {
+			var err2 error
+			if err != nil {
+				err2 = tx.Rollback(ctx)
+				if err2 != nil {
+					err = fmt.Errorf("rollback txn: %w (while processing %s)", err2, err)
+				}
+			} else {
+				err2 = tx.Commit(ctx)
+				if err2 != nil {
+					err = fmt.Errorf("commit txn: %w", err2)
+				}
+			}
+		}()
+		for i := 0; i < len(batch.Events); i++ {
+			event := batch.Events[i]
+			stmt := event.GetSQLStmt(yb.tconf.Schema)
+			log.Debug(stmt)
+			_, err := tx.Exec(context.Background(), stmt)
 			if err != nil {
 				log.Errorf("Error executing stmt: %v", err)
+				return false, fmt.Errorf("error executing stmt - %v: %w", stmt, err)
 			}
-			return false, err
-		})
-		if err != nil {
-			return fmt.Errorf("error executing stmt - %v: %w", stmt, err)
 		}
+		importStateQuery := fmt.Sprintf(`UPDATE %s SET last_applied_vsn=%d where migration_uuid='%s' AND channel_no=%d;`, EVENT_CHANNELS_METADATA_TABLE_NAME, batch.GetLastVsn(), migrationUUID, batch.ChanNo)
+		res, err := tx.Exec(context.Background(), importStateQuery)
+		res.RowsAffected()
```

Comment:
Do you want to log it?

Reply:
oops, missed removing it. Added a log anyway.